compiler/qsc_doc_gen/src/generate_docs.rs

```diff
@@ -189,101 +320,103 @@ pub fn generate_docs(
 
         let package = &unit.package;
         for (_, item) in &package.items {
-            if let Some((ns, line)) = generate_doc_for_item(
+            if let Some((ns, metadata)) = generate_doc_for_item(
+                package_id,
                 package,
                 package_kind.clone(),
                 is_current_package,
                 item,
                 display,
                 &mut files,
             ) {
-                toc.entry(ns).or_default().push(line);
+                toc.entry(ns).or_default().push(metadata);
             }
         }
     }
 
+    // Generate Overview files for each namespace
+    for (ns, items) in &mut toc {
+        generate_index_file(&mut files, ns, items);
+    }
+
     // We want to sort documentation files in a meaningful way.
     // First, we want to put files for the current project, if it exists.
     // Then we want to put explicit dependencies of the current project, if they exist.
     // Then we want to add built-in std package. And finally built-in core package.
     // Namespaces within packages should be sorted alphabetically and
-    // items with a namespace should be also sorted alphabetically.
+    // items with a namespace should be also sorted alphabetically with the index file appearing first.
     // Also, items without any metadata (table of content) should come last.
     files.sort_by_key(|file| {
-        (
-            file.1.package.clone(),
-            file.1.namespace.clone(),
-            file.1.name.clone(),
-        )
+        let prefix = if file.0.ends_with("index.md") {
+            "0"
```

Comment:
can this be an int, so the sorting order is more clear? (not sure if stringified numbers always sort in the order we'd expect)